### PR TITLE
Cluster config, Failure group and Replica changes

### PR DESCRIPTION
**Pull request by jayeshh123**
This PR is related to cluster config, failure group and replica changes

### CODE REVIEW

File: resources/common/network_playbook/network_playbook.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+*/
+
+variable "compute_cluster_create_complete" {}
+variable "storage_cluster_create_complete" {}
+variable "remote_mount_create_complete" {}
+variable "inventory_path" {}
+variable "network_playbook_path" {}
+
+resource "null_resource" "perform_scale_deployment" {
+  count = (tobool(var.compute_cluster_create_complete) == true && tobool(var.storage_cluster_create_complete) == true && tobool(var.remote_mount_create_complete) == true) ? 1 : 0
```
Comment (digvijay-ukirde) [inline resources/common/network_playbook/network_playbook.tf]: `count = (tobool(var.turn_on) == true && tobool(var.clone_complete) == true && tobool(var.compute_cluster_create_complete) == true && tobool(var.storage_cluster_create_complete) == true && tobool(var.create_scale_cluster) == true) ? 1 : 0`
remote_mount_create_complete is not a must have

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf [hunk 2]
```diff
@@ -588,6 +594,26 @@ module "remote_mount_configuration" {
   depends_on                      = [module.gklm_instance, module.compute_cluster_configuration, module.storage_cluster_configuration, module.combined_cluster_configuration]
 }
 
+module "invoke_compute_network_playbook" {
+  source                          = "../../../resources/common/network_playbook"
+  compute_cluster_create_complete = module.compute_cluster_configuration.compute_cluster_create_complete
+  storage_cluster_create_complete = module.storage_cluster_configuration.storage_cluster_create_complete
+  remote_mount_create_complete    = module.remote_mount_configuration.remote_mount_create_complete
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: remove

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf [hunk 3]
```diff
@@ -588,6 +594,26 @@ module "remote_mount_configuration" {
...
+  inventory_path                  = format("%s/%s/compute_inventory.ini", var.scale_ansible_repo_clone_path, "ibm-spectrum-scale-install-infra")
+  network_playbook_path           = format("%s/%s/collections/ansible_collections/ibm/spectrum_scale/samples/playbook_cloud_network_config.yaml", var.scale_ansible_repo_clone_path, "ibm-spectrum-scale-install-infra")
+  depends_on                      = [module.remote_mount_configuration]
+}
+
+module "invoke_storage_network_playbook" {
+  source                          = "../../../resources/common/network_playbook"
+  compute_cluster_create_complete = module.compute_cluster_configuration.compute_cluster_create_complete
+  storage_cluster_create_complete = module.storage_cluster_configuration.storage_cluster_create_complete
+  remote_mount_create_complete    = module.remote_mount_configuration.remote_mount_create_complete
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: remove

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf [hunk 4]
```diff
@@ -588,6 +594,26 @@ module "remote_mount_configuration" {
...
 }
 
+module "invoke_compute_network_playbook" {
+  source                          = "../../../resources/common/network_playbook"
+  compute_cluster_create_complete = module.compute_cluster_configuration.compute_cluster_create_complete
+  storage_cluster_create_complete = module.storage_cluster_configuration.storage_cluster_create_complete
+  remote_mount_create_complete    = module.remote_mount_configuration.remote_mount_create_complete
+  inventory_path                  = format("%s/%s/compute_inventory.ini", var.scale_ansible_repo_clone_path, "ibm-spectrum-scale-install-infra")
+  network_playbook_path           = format("%s/%s/collections/ansible_collections/ibm/spectrum_scale/samples/playbook_cloud_network_config.yaml", var.scale_ansible_repo_clone_path, "ibm-spectrum-scale-install-infra")
+  depends_on                      = [module.remote_mount_configuration]
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: The network playbook shouldn't depend on the remote mount. We should be able to run it once inventories are available.

File: resources/common/network_playbook/network_playbook.tf (hunk 1 above)
Comment (jayeshh123) [reply resources/common/network_playbook/network_playbook.tf]: Resolved.

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf (hunk 2 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: Resolved.

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf (hunk 3 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: Resolved.

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf (hunk 4 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: Resolved.

**Review by sasikeda — APPROVED** [review]

**Pull request merged**